docs/resources/Auto_Moderation.md

```diff
@@ -0,0 +1,184 @@
+  ],
+  "trigger_metadata": {
+    "keyword_filter": ["cat*", "*cat", "*ana*"]
+  },
+  "enabled": true,
+  "exempt_roles": ["323456789123456789", "423456789123456789"],
+  "exempt_channels": ["523456789123456789"],
+}
+```
+
+###### Trigger Types
+
+Characterizes what type of information will be checked to determine whether a rule is triggered.
+
+| Type                 | Value   | Description                                                                     |
+| -------------------- | ------- | ------------------------------------------------------------------------------- |
+| KEYWORDS             | 1       | check if any words from a user defined list of keywords exist in content        |
+| HARMFUL_LINKS        | 2       | check if any known harmful links exist in content                               |
+| SPAM                 | 3       | check if content represents generic spam                                        |
+| DEFAULT_KEYWORD_LIST | 4       | check if any words from built in pre-determined lists of words exist in content |
+
+###### Trigger Metadata
+
+Additional data needed to figure out whether a rule should be triggered. Different fields are relevant based on the
+value of [trigger_type](#DOCS_RESOURCES_AUTO_MODERATION/auto-moderation-rule-object-trigger-types).
+
+**TODO**
+
+###### Event Types
+
+Indicates in what event context a rule should be checked.
+
+| Type             | Value   | Description                              |
+| ---------------- | ------- | ---------------------------------------- |
+| MESSAGE_SEND     | 1       | when a member sends a message on a guild |
+
+
+### Auto Moderation Action Object
+
+An action which will execute whenever a rule is triggered.
+
+###### Auto Moderation Action Structure
+
+| Field                     | Type                                                                                      | Description                                                               |
+| ------------------------- | ----------------------------------------------------------------------------------------- | ------------------------------------------------------------------------- |
+| type                      | [action type](#DOCS_RESOURCES_AUTO_MODERATION/auto-moderation-action-object-action-types) | the type of action                                                        |
+| metadata                  | **TODO**                                                                                  | additional metadata needed during execution for this specific action type |
+
+
+###### Action Types
+
+| Type               | Value   | Description                                                     |
+| ------------------ | ------- | --------------------------------------------------------------- |
+| BLOCK_MESSAGE      | 1       | blocks the content of a message according to the rule           |
+| LOG_TO_CHANNEL     | 2       | records original message in a specified channel                 |
```

Comment:
internally we've been naming this `FLAG_TO_CHANNEL`. But i think we should improve on the name. Flag is a bit ambiguous so I'm thinking of the *log* term instead. And instead of LOG_TO_CHANNEL, maybe we just use `LOG_MESSAGE`? Mentioning channel in name seems unecessary and it'll be consistent with BLOCK_MESSAGE. What do ya'll think 🤔

Reply:
I agree, but I was thinking about another idea and I don't know if it can be `LOG_MESSAGE` or `MESSAGE_LOG`. I'm hesitating a little bit because you mentioned that LOG_MESSAGE will be consistent with BLOCK_MESSAGE. It's only an idea. What do you think?

Reply:
maybe `SEND_ALERT` to be consistent with the UI?

Reply:
Could be, good idea

Reply:
not the biggest fan of the "alert" term here for this use case (even in the frontend) but will change it to that for now for consistency with the frontend 👍  good suggestion